vortex-array/src/vtable/statistics.rs

```diff
@@ -22,3 +23,57 @@ where
         StatisticsVTable::compute_statistics(encoding, array_ref, stat)
     }
 }
+
+impl Array {
+    /// Computes ths statistics for the given array and stat. This will update the stats of the array
+    /// and return this set.
+    ///
+    /// Other stats might be computed or inferred at the same time.
+    pub fn compute_statistics(&self, stat: Stat) -> VortexResult<StatsSet> {
```

Comment:
Wait.. isn't this currently `array.statistics().compute()`? What's the difference? Should we deprecate that other trait in favor of just calling this function? 

I think we meant to remove the trait after refactoring StatsSet at some point

Reply:
I think so, the point of this is that at each rec level it get called again can house stat -> stat inference.

Reply:
We can likely remove this, but I wanted to not touch the world again, i will follow up.

Reply:
can remove `compute_all`?

Reply:
Or do you think that impl Statistics goes and we use a stats set instead?. I don't think that affects the underlying array, maybe impl Statistics, but becomes `set` and `clear`